jupyterhubutils/lsstmgr/namespacemanager.py

```diff
@@ -0,0 +1,298 @@
+'''Class to provide namespace manipulation.
+'''
+
+import time
+from kubernetes.client.rest import ApiException
+from kubernetes import client
+
+from ..utils import (get_execution_namespace, make_logger)
+
+
+class LSSTNamespaceManager(object):
+    '''Class to provide namespace manipulation.
+    '''
+    namespace = None
+    service_account = None
+
+    def __init__(self, *args, **kwargs):
+        self.log = make_logger()
+        self.log.debug("Creating LSSTNamespaceManager")
+        self.parent = kwargs.pop('parent')
+
+    def update_namespace_name(self):
+        '''Build namespace name from user and execution namespace.
+        '''
+        execution_namespace = get_execution_namespace()
+        self.log.debug("Execution namespace: '{}'".format(execution_namespace))
+        user = self.parent.user
+        self.log.debug("User: '{}'".format(user))
+        username = user.escaped_name
+        if execution_namespace and username:
+            self.namespace = "{}-{}".format(execution_namespace,
+                                            username)
+        else:
```

Comment:
If you don't have a namespace just throw don't try to shove everything in default